adapters/Unity/UnityAdapter/GADMAdapterUnityRewardedAd.m

```diff
@@ -48,191 +43,138 @@ @implementation GADMAdapterUnityRewardedAd
 
 - (instancetype)initWithAdConfiguration:(GADMediationRewardedAdConfiguration *)adConfiguration
                       completionHandler:
-                          (GADMediationRewardedLoadCompletionHandler)completionHandler {
-  self = [super init];
-  if (self) {
-    _adLoadCompletionHandler = completionHandler;
-    _adConfiguration = adConfiguration;
-
-    _uuid = [[NSUUID UUID] UUIDString];
-
-    _metaData = [[UADSMetaData alloc] init];
-    [_metaData setCategory:@"mediation_adapter"];
-    [_metaData set:_uuid value:@"create-adapter"];
-    [_metaData commit];
-  }
-  return self;
+(GADMediationRewardedLoadCompletionHandler)completionHandler {
+    self = [super init];
+    if (self) {
+        _adLoadCompletionHandler = completionHandler;
+        _adConfiguration = adConfiguration;
+    }
+    return self;
 }
 
 - (void)requestRewardedAd {
-  _gameID = [_adConfiguration.credentials.settings objectForKey:kGADMAdapterUnityGameID];
-  _placementID = [_adConfiguration.credentials.settings objectForKey:kGADMAdapterUnityPlacementID];
-  NSLog(@"Requesting unity rewarded ad with placement: %@", _placementID);
-
-  GADMAdapterUnityRewardedAd __weak *weakSelf = self;
-
-  if (!_gameID || !_placementID) {
-    if (_adLoadCompletionHandler) {
-      NSError *error = GADMAdapterUnityErrorWithCodeAndDescription(
-          GADMAdapterUnityErrorInvalidServerParameters, @"Game ID and Placement ID cannot be nil.");
-      _adLoadCompletionHandler(nil, error);
-      _adLoadCompletionHandler = nil;
+    _gameID = [_adConfiguration.credentials.settings objectForKey:kGADMAdapterUnityGameID];
+    _placementID = [_adConfiguration.credentials.settings objectForKey:kGADMAdapterUnityPlacementID];
+    NSLog(@"Requesting unity rewarded ad with placement: %@", _placementID);
+    if (!_gameID || !_placementID) {
+        if (_adLoadCompletionHandler) {
+            NSError *error = GADMAdapterUnityErrorWithCodeAndDescription(GADMAdapterUnityErrorInvalidServerParameters, kMISSING_ID_ERROR);
+            _adLoadCompletionHandler(nil, error);
+            _adLoadCompletionHandler = nil;
+        }
+        return;
     }
-    return;
-  }
-
-  if (![UnityAds isSupported]) {
-    NSString *description =
+    
+    if (![UnityAds isSupported]) {
```

Comment:
maybe move this inside the initializer - seems strange for this to be the only place we check this

Reply:
it can't be moved into the Initializer because inside of the if statements we are using the _adLoadCompletionHandler, but any thoughts on if this block is really necessary, since the Initializer will check if Unity ads is supported for the device?

Reply:
you can move it. You can do very similar thing as on Android (https://github.com/Applifier/googleads-mobile-android-mediation/blob/fabd60c33ce95322baa1042132e8b7b9e273ef5f/ThirdPartyAdapters/unity/unity/src/main/java/com/google/ads/mediation/unity/UnityInitializer.java#L60). `_adLoadCompletionHandler` is needed only in context of `requestRewardedAd`.

I would agree that `if (![UnityAds isSupported])` should not be here, but we might want to keep here for now.

Reply:
it all depend on AdMob. Can admob call requestRewardedAd before we our SDK got initialized through adapter or not.